### PR TITLE
python3Packages.collections-extended: init at 2.0.0

### PR DESCRIPTION
**Pull request by exarkun**
###### Motivation for this change

An upcoming release of nixos.tahoe-lafs has this library as a new dependency.

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/collections-extended/default.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256:16vyx8c5glqdyzjjpfwqp1cx8jlcshh0pi85n9mg8475ah07h7b5";
+  };
+
+  # As of 2.0.0 the tests aren't in the distribution archive.
+  doCheck = false;
+
+  meta = with lib; {
+    homepage = https://github.com/mlenzen/collections-extended;
```
Comment (smancill) [inline pkgs/development/python-modules/collections-extended/default.nix:21]: The url needs to be a string (quoted).

File: pkgs/development/python-modules/collections-extended/default.nix [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+  disabled = pythonOlder "3.6";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256:16vyx8c5glqdyzjjpfwqp1cx8jlcshh0pi85n9mg8475ah07h7b5";
+  };
+
+  # As of 2.0.0 the tests aren't in the distribution archive.
+  doCheck = false;
```
Comment (smancill) [inline pkgs/development/python-modules/collections-extended/default.nix:18]: I've been asked in previous PR to switch to the GitHub archive if the tests are not available in the PyPi release, and use `pythonImportsCheck`.

**Comment by exarkun** [comment]
Thanks for the feedback @smancill .  I think I've addressed those points.

**Comment by smancill** [comment]
Result of `nixpkgs-review pr 140094` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python38Packages.collections-extended</li>
    <li>python39Packages.collections-extended</li>
  </ul>
</details>

**Comment by smancill** [comment]
Works fine.

**Review by risicle — APPROVED** [review]
LGTM

`nixpkgs-review` happy on macos 10.15 & nixos x86_64

**Pull request merged**